### PR TITLE
MYR-119 : Fix tests disabled by MYR-15

### PR DESCRIPTION
**Pull request by george-lorch**
MYR-58 : 5.7 mtr - re-record tests for Warning 1287 - deprecated partition
         handler
MYR-59 : 5.7 mtr - fix and re-record tests for deprecated EXPLAIN EXTENDED,
         new EXPLAIN result format, and format=json
MYR-61 : 5.7 mtr - fix and re-record tests using deprecated
         i_s.global_status
MYR-79 : 5.7 mtr - re-record tests that use restart_mysqld.inc due to new
         comment in output
MYR-81 : 5.7 mtr - re-record tests for new warning on INSERT IGNORE when
         duplicate exists
MYR-99 : Convert tests to not require have_rocksdb_as_default
- Many tests were disabled by MYR-15 due to new gap lock detection and SQL
  error. Tests needed some work to use either different queries or to
  temporarily switch ISO level to READ-COMMITTED
- While fixing tests, a few new issues appeared and were reported:
  - MYR-122 : split up rocksdb.rocksdb test into individual tests that
    actually test something specific
  - MYR-123 : PS 5.7 rocksdb_max_open_files differs from default
- Other issues listed above were addressed/re-recorded as impacted tests
  were enabled.
- A couple of the tests required some more significant work in order to
  test original intention and variant results based on ISO mode and
  intersection with gap lock error in RR ISO mode.
  - rocksdb.transaction reimplemented to some extent to run the full test
    once under each supported MyRocks ISO mode (READ COMMITTED and
    REPEATABLE READ)
  - rocksdb.unique_sec reimplemented to run the full test once under each
    supported MyRocks ISO mode (READ COMMITTED and REPEATABLE READ)
  - rocksdb.unique_sec_rev_cf reimplemented to run the full test once under
    each supported MyRocks ISO mode (READ COMMITTED and REPEATABLE READ)
- The rocksdb.rocksdb test is a special case as noted by the new issue
  above and just needs to be cratered and started over.

### CODE REVIEW

**Comment by george-lorch** [comment]
http://jenkins.percona.com/view/5.7/job/mysql-5.7-param/825/

File: mysql-test/suite/rocksdb/include/locking_issues_case1_1.inc [hunk 1]
```diff
@@ -17,11 +17,7 @@
 --echo -   using $isolation_level transaction isolation level
 --echo -----------------------------------------------------------------------
 
---disable_warnings
-DROP TABLE IF EXISTS t0;
---enable_warnings
-
-CREATE TABLE t0(id1 INT, id2 INT, value INT, PRIMARY KEY(id1, id2));
+CREATE TABLE t0(id1 INT, id2 INT, value INT, PRIMARY KEY(id1, id2)) engine=rocksdb;
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/include/locking_issues_case1_1.inc]: Here and everywhere inconsistent SQL syntax case "CREATE" but "engine". I don't know the official RocksDB capitalisation for the plugin SE name, likely it's "RocksDB"

File: mysql-test/suite/rocksdb/t/delete.test [hunk 2]
```diff
@@ -7,9 +7,10 @@
 # DELETE IGNORE is covered in delete_ignore test
 #
 
---disable_warnings
-DROP TABLE IF EXISTS t1,t2;
---enable_warnings
+# MyRocks does not support gap locks in REPEATABLE-READ mode, test does not
+# require RR ISO to complete, so lets temporarily alter the ISO to RC
+set @orig_tx_iso=@@session.tx_isolation;
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/t/delete.test]: Here and everywhere, no need to save/restore the session variable, when it's restored at the very end of a testcase

File: mysql-test/suite/rocksdb/t/rocksdb.test [hunk 3]
```diff
@@ -825,6 +829,7 @@ send update t49 set a = 1000 where pk = 1;
 --connect (con2,localhost,root,,)
 --echo kill query \$con1_id;
 --disable_query_log
+# this is bad and will result in unstable tests
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/t/rocksdb.test]: Add some TODO marker?

File: mysql-test/suite/rocksdb/t/rocksdb.test [hunk 4]
```diff
@@ -1083,12 +1108,17 @@ insert into t1 values (1,10),(2,20);
 --let $con_id = `select connection_id()`
 
 set autocommit = 1;
---send
-update t1 set a = sleep(100) where pk = 1;
+# this is bad and will result in unstable tests
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/t/rocksdb.test]: Likewise here and everywhere

File: mysql-test/suite/rocksdb/t/select_for_update.test [hunk 5]
```diff
@@ -1,21 +1,29 @@
...
+set session tx_isolation='READ-COMMITTED';
+
 CREATE TABLE t1 (a INT, b CHAR(8), pk INT AUTO_INCREMENT PRIMARY KEY) ENGINE=rocksdb;
 INSERT INTO t1 (a,b) VALUES (1,'a'),(2,'b'),(3,'a');
 
 --connect (con1,localhost,root,,)
+# MyRocks does not support gap locks in REPEATABLE-READ mode, this test
+# does not require RR ISO to complete, so lets temporarily alter the ISO
+# to RC
+set @orig_tx_iso=@@session.tx_isolation;
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/t/select_for_update.test]: Saving the session variable is twice as much not required, when you just opened a session :)

File: mysql-test/suite/rocksdb/t/transaction.test [hunk 6]
```diff
@@ -35,40 +35,112 @@ update t1 set id=100 where id=1;
 rollback;
 
 # update non-index column key by secondary key
+# MyRocks does not support gap locks in REPEATABLE-READ mode, so we will
+# execute this twice, once to illustrate the gap lock error in RR mode, and once
+# in RC mode to illistrate passing functionality
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/t/transaction.test]: s/illistrate/illiustrate/g

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

File: mysql-test/suite/rocksdb/include/locking_issues_case1_1.inc (hunk 1 above)
Comment (george-lorch) [reply mysql-test/suite/rocksdb/include/locking_issues_case1_1.inc]: Yeah, case consistency is non existent in this suite. In general, I should have been consistent with the case of the statement.

File: mysql-test/suite/rocksdb/t/delete.test (hunk 2 above)
Comment (george-lorch) [reply mysql-test/suite/rocksdb/t/delete.test]: True, end of tests really don't necessarily need to reset sessions, but, IIRC, I have been requested to track and reset sessions in reviews before. Perhaps it was not a realization that a set/reset was session only or that maybe future expectation is that even session sight span test cases. I'll remove them and yell at the next reviewer that asks that I reset a session var to its original value.

File: mysql-test/suite/rocksdb/t/rocksdb.test (hunk 3 above)
Comment (george-lorch) [reply mysql-test/suite/rocksdb/t/rocksdb.test]: I thought this in itself was a marker, along with new MYR issue that this test needs to be destroyed and started over.

File: mysql-test/suite/rocksdb/t/transaction.test (hunk 6 above)
Comment (george-lorch) [reply mysql-test/suite/rocksdb/t/transaction.test]: Stupid sausage fingers

File: mysql-test/suite/rocksdb/t/rocksdb.test (hunk 3 above)
Comment (laurynas-biveinis) [reply mysql-test/suite/rocksdb/t/rocksdb.test]: OK then

**Comment by george-lorch** [comment]
Changes made and re-jenkinsed http://jenkins.percona.com/view/5.7/job/mysql-5.7-param/833/

**Review by laurynas-biveinis — APPROVED** [review]

**Pull request merged**